Failure while resolving values of package private annotations.

OGM fails to read the values of package private annotations when such an annotation is used on a scanned class. A runtime exception is cause by

```
Caused by: java.lang.IllegalAccessException: class org.neo4j.ogm.metadata.AnnotationInfo cannot access a member of interface XXX with modifiers "public abstract".
```

--- a/core/src/main/java/org/neo4j/ogm/metadata/AnnotationInfo.java
+++ b/core/src/main/java/org/neo4j/ogm/metadata/AnnotationInfo.java
@@ -21,6 +21,8 @@ package org.neo4j.ogm.metadata;
 import java.lang.annotation.Annotation;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
 import java.util.HashMap;
 import java.util.Map;
 
@@ -88,13 +90,15 @@ public class AnnotationInfo {
     }
 
     private static Object getAttributeValue(Annotation annotation, Method element) {
-        Object value;
-        try {
-            value = element.invoke(annotation);
-        } catch (IllegalAccessException | InvocationTargetException e) {
-            throw new RuntimeException("Could not read value of Annotation " + element.getName(), e);
-        }
-        return value;
+
+        return AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
+            try {
+                element.setAccessible(true);
+                return element.invoke(annotation);
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                throw new RuntimeException("Could not read value of Annotation " + element.getName(), e);
+            }
+        });
     }
 
     public String getName() {

--- a/core/src/test/java/org/neo4j/ogm/domain/gh827/DomainObject.java
+++ b/core/src/test/java/org/neo4j/ogm/domain/gh827/DomainObject.java
@@ -16,12 +16,19 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.neo4j.ogm.metadata.schema.gh670;
+package org.neo4j.ogm.domain.gh827;
 
+import org.neo4j.ogm.annotation.GeneratedValue;
+import org.neo4j.ogm.annotation.Id;
 import org.neo4j.ogm.annotation.NodeEntity;
 
+/**
+ * @author Michael J. Simons
+ */
 @NodeEntity
-public class Course {
+public class DomainObject {
+
+    @Id @GeneratedValue
     private Long id;
-    private Long title;
+
 }

--- a/core/src/test/java/org/neo4j/ogm/domain/gh827/PackagePrivateAnnotation.java
+++ b/core/src/test/java/org/neo4j/ogm/domain/gh827/PackagePrivateAnnotation.java
@@ -1,0 +1,36 @@
+/*
+ * Copyright (c) 2002-2020 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.ogm.domain.gh827;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * @author Michael J. Simons
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@interface PackagePrivateAnnotation {
+
+    String value() default "";
+}

--- a/core/src/test/java/org/neo4j/ogm/domain/gh827/UnrelatedClass.java
+++ b/core/src/test/java/org/neo4j/ogm/domain/gh827/UnrelatedClass.java
@@ -16,13 +16,11 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.neo4j.ogm.metadata.schema.gh670;
+package org.neo4j.ogm.domain.gh827;
 
-import java.util.List;
-
-import org.neo4j.ogm.annotation.NodeEntity;
-
-@NodeEntity
-public class Teacher extends Person {
-    private List<Course> cources;
+/**
+ * @author Michael J. Simons
+ */
+@PackagePrivateAnnotation
+public class UnrelatedClass {
 }

--- a/core/src/test/java/org/neo4j/ogm/metadata/AnnotationsInfoTest.java
+++ b/core/src/test/java/org/neo4j/ogm/metadata/AnnotationsInfoTest.java
@@ -16,12 +16,20 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.neo4j.ogm.metadata.schema.gh670;
+package org.neo4j.ogm.metadata;
 
-import org.neo4j.ogm.annotation.NodeEntity;
+import org.junit.Test;
+import org.neo4j.ogm.domain.gh827.UnrelatedClass;
 
-@NodeEntity
-public class Person {
-    private Long id;
-    private String name;
+/**
+ * @author Michael J. Simons
+ */
+public class AnnotationsInfoTest {
+
+    @Test
+    public void shouldBeAbleToHandlePackagePrivateAnnotations() {
+
+        AnnotationsInfo info = new AnnotationsInfo(UnrelatedClass.class);
+
+    }
 }

